docs.joern.io/content/frontends/_index.md

```diff
@@ -0,0 +1,41 @@
+---
+id: frontends 
+title: Frontends
+bookCollapseSection: true
+weight: 135
+---
+
+## Frontends
+
+Joern contains multiple frontends, each responsible for parsing the source code and generating the AST for their respective languages. `X2CPG` is the base on which all of the other frontends are built, and contains common structures by all of the languages. Each frontend contains the base arguments from `x2cpg`, as well as custom arguments that is frontend specific. See each respective languages' page for a detailed explanation on what arguments are available and what each of the arguments do.
+
+## Invoking a specific frontend
+To generate the AST for source files with a specific frontend you will need to find the frontend in the `joern-cli/frontends/` folder. Once there, you will need to build the frontend using `sbt` afterwhich you can invoke it direclty with the script that is generated. As an example, to parse a given Python directory and generate an AST:
```

Comment:
"To generate the AST for source files with a specific frontend you will need to find the frontend in the..." => "To generate the AST for some program with a specific frontend, invoke the desired frontend from the `joern-cli/frontends` directory using the generated script. If the script is not there, make sure that Joern has been staged with `sbt`. As an example...."